app/templating/template_pre_processor.py

```diff
@@ -46,7 +46,17 @@ def get_template_name(self):
             return 'questionnaire.html'
 
     def build_view_data(self):
-        self._augment_questionnaire()
+        # Always plumb the questionnaire
+        self._plumb_questionnaire()
+
```

Comment:
These 2 if statements are related, both requiring the user to be in the questionnaire, the only difference is the 2nd one just excludes the summary page. So I would consider

       # We only need the answers if we are in the questionnaire
        if current_location != 'introduction' and current_location != 'thank-you':
            self._collect_answers()

            # We don't need to augment the summary page
            if current_location != 'summary':
                self._augment_questionnaire()